### ISSUE TITLE
Borrow visualizer

### ISSUE THREAD

**Issue by kindlychung** [intellij-rust/intellij-rust#968]
See the discussion and prototype here: 

https://internals.rust-lang.org/t/borrow-visualizer-for-the-rust-language-service/4187/28

Would be cool if we had this in intellij-rust!

**Comment by farodin91** [comment]
@matklad I would like to work on something like this have you any idea?

**Comment by matklad** [comment]
I don't think we have the necessary infrastructure in place to implement this reliably at the moment unfortunately.

**Comment by vlad20012** [comment]
The main infrastructure for anything related to borrower is the type inference. But not the type inference we currently working on. It will seem like a child's toy in comparison with the real rustc's type inference (with it's subtyping, variances, relations and so on) that infer lifetime parameters.
